core/services/keeper/upkeep_executer.go

```diff
@@ -29,10 +29,22 @@ const (
 	executionQueueSize = 10
 )
 
-// UpkeepExecuter fulfills Service and HeadBroadcastable interfaces
-var _ job.Service = (*UpkeepExecuter)(nil)
-var _ httypes.HeadTrackable = (*UpkeepExecuter)(nil)
+// Revert reasons
+const (
+	UpkeepNotNeededReason RevertReason = "upkeep not needed"
+	OutOfTurnReason       RevertReason = "keepers must take turns"
```

Comment:
the following are all common, expected errors messages that don't signal an issue with the protocol. They can be debug level.
* "insufficient funds"
* "call to perform upkeep failed"
* "call to check target failed: <custom>"

Reply:
Done